### ISSUE TITLE
doesn't work on macOS

### ISSUE THREAD

**Issue by bsideup** [gorkem/vscode-java#32]
Java 8 is installed, but can't be recognized. Links to RedHat's OpenJDK, but it has no macOS version as well.

**Comment by ianrobrien** [comment]
@bsideup is your `JAVA_HOME` environment variable set and pointing to the 1.8 JDK (top level directory)? That's the issue I ran into.

**Comment by bsideup** [comment]
@ianrobrien yeah, I think that's the reason. IMO JAVA_HOME should be mentioned next to "download OpenJDK" with something like "and make sure your JAVA_HOME is set"

**Comment by tills13** [comment]
^ I agree. Especially considering you **don't actually need OpenJDK**. The error message is kind of pointing fingers when it really doesn't need to be.

**Comment by fbricon** [comment]
OpenJDK is one possible implementation for the JDK. Runs fine on Oracle.

The d/l link should definitely not point to the OpenJDK d/l page. Probably default to the Oracle one in that case.

**Comment by jglete** [comment]
I have put a `~/.bashrc` with this:

```
export JAVA_HOME=$(/usr/libexec/java_home -v 1.8)
export PATH=$JAVA_HOME/bin:$PATH
```

For this test:

```
. ~/.bashrc
open -a "Visual Studio Code"
```

This way I'm sure this time the JAVA_HOME variable is set.

Now I create a new file, change the language to java an then pops this stacktrace:

```
[Error] 22-sep-2016 23:17:02 Exception on incoming message dispatcher
null argument:
org.eclipse.core.runtime.AssertionFailedException: null argument:
	at org.eclipse.core.runtime.Assert.isNotNull(Assert.java:85)
	at org.eclipse.core.runtime.Assert.isNotNull(Assert.java:73)
	at org.eclipse.core.runtime.Path.initialize(Path.java:641)
	at org.eclipse.core.runtime.Path.<init>(Path.java:238)
	at org.eclipse.core.runtime.Path.<init>(Path.java:186)
	at org.eclipse.core.runtime.Path.fromOSString(Path.java:92)
	at org.jboss.tools.vscode.java.handlers.AbstractRequestHandler.resolveCompilationUnit(AbstractRequestHandler.java:33)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler.handleOpen(DocumentLifeCycleHandler.java:127)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler.access$1(DocumentLifeCycleHandler.java:126)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler$OpenHandler$1.run(DocumentLifeCycleHandler.java:75)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2240)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2262)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler$OpenHandler.handle(DocumentLifeCycleHandler.java:72)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler$OpenHandler.handle(DocumentLifeCycleHandler.java:1)
	at org.jboss.tools.langs.base.LSPServer.dispatchNotification(LSPServer.java:222)
	at org.jboss.tools.langs.base.LSPServer.messageReceived(LSPServer.java:156)
	at org.jboss.tools.langs.transport.AbstractConnection$DispatcherThread.run(AbstractConnection.java:32)

[Error] 22-sep-2016 23:17:02 Exception on incoming message dispatcher
null argument:
org.eclipse.core.runtime.AssertionFailedException: null argument:
	at org.eclipse.core.runtime.Assert.isNotNull(Assert.java:85)
	at org.eclipse.core.runtime.Assert.isNotNull(Assert.java:73)
	at org.eclipse.core.runtime.Path.initialize(Path.java:641)
	at org.eclipse.core.runtime.Path.<init>(Path.java:238)
	at org.eclipse.core.runtime.Path.<init>(Path.java:186)
	at org.eclipse.core.runtime.Path.fromOSString(Path.java:92)
	at org.jboss.tools.vscode.java.handlers.AbstractRequestHandler.resolveCompilationUnit(AbstractRequestHandler.java:33)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler.handleChanged(DocumentLifeCycleHandler.java:155)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler.access$2(DocumentLifeCycleHandler.java:154)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler$ChangeHandler$1.run(DocumentLifeCycleHandler.java:98)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2240)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2262)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler$ChangeHandler.handle(DocumentLifeCycleHandler.java:95)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler$ChangeHandler.handle(DocumentLifeCycleHandler.java:1)
	at org.jboss.tools.langs.base.LSPServer.dispatchNotification(LSPServer.java:222)
	at org.jboss.tools.langs.base.LSPServer.messageReceived(LSPServer.java:156)
	at org.jboss.tools.langs.transport.AbstractConnection$DispatcherThread.run(AbstractConnection.java:32)

[Error] 22-sep-2016 23:17:02 Exception on incoming message dispatcher
null argument:
org.eclipse.core.runtime.AssertionFailedException: null argument:
	at org.eclipse.core.runtime.Assert.isNotNull(Assert.java:85)
	at org.eclipse.core.runtime.Assert.isNotNull(Assert.java:73)
	at org.eclipse.core.runtime.Path.initialize(Path.java:641)
	at org.eclipse.core.runtime.Path.<init>(Path.java:238)
	at org.eclipse.core.runtime.Path.<init>(Path.java:186)
	at org.eclipse.core.runtime.Path.fromOSString(Path.java:92)
	at org.jboss.tools.vscode.java.handlers.AbstractRequestHandler.resolveCompilationUnit(AbstractRequestHandler.java:33)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler.handleChanged(DocumentLifeCycleHandler.java:155)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler.access$2(DocumentLifeCycleHandler.java:154)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler$ChangeHandler$1.run(DocumentLifeCycleHandler.java:98)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2240)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2262)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler$ChangeHandler.handle(DocumentLifeCycleHandler.java:95)
	at org.jboss.tools.vscode.java.handlers.DocumentLifeCycleHandler$ChangeHandler.handle(DocumentLifeCycleHandler.java:1)
	at org.jboss.tools.langs.base.LSPServer.dispatchNotification(LSPServer.java:222)
	at org.jboss.tools.langs.base.LSPServer.messageReceived(LSPServer.java:156)
	at org.jboss.tools.langs.transport.AbstractConnection$DispatcherThread.run(AbstractConnection.java:32)


```

**Comment by fbricon** [comment]
@jglete that's a separate issue, see #38. Currently the Java Language Server only supports opening existing Maven-based projects and to a lesser degree, simple Eclipse projects with the Java nature.

As for the original problem, I consider it fixed along #37. The error message is now a bit more explicit:
![](http://content.screencast.com/users/fbricon/folders/Jing/media/9c531f13-934e-4119-b483-fb481031e600/00000468.png)
